File: mynlp-segment/src/main/java/com/mayabot/nlp/segment/reader/StopWordDict.java
package com.mayabot.nlp.segment.reader;

import com.google.common.collect.Sets;
import com.google.inject.Inject;
import com.google.inject.Singleton;
import com.mayabot.nlp.MynlpEnv;
import com.mayabot.nlp.resources.NlpResource;
import com.mayabot.nlp.utils.CharSourceLineReader;

import java.util.Set;

/**
 * 停用词词典
 *
 * @author jimichan
 */
@Singleton
public class StopWordDict {

    private Set<String> stopWords;


    @Inject
    public StopWordDict(MynlpEnv env) {

        Set<String> set = Sets.newHashSet();

        NlpResource resource = env.loadResource("stopword-dict/stopwords.txt");

        try (CharSourceLineReader reader = resource.openLineReader()) {
            while (reader.hasNext()) {

                String line = reader.next();

                set.add(line.trim());
            }
        } catch (Exception e) {
            throw new RuntimeException(e);
        }

        this.stopWords = set;
    }

    public Set<String> getStopWords() {
        return stopWords;
    }

    public StopWordDict setStopWords(Set<String> stopWords) {
        this.stopWords = stopWords;
        return this;
    }
}

File: mynlp-segment/src/main/java/com/mayabot/nlp/segment/reader/StopWordDict.kt
package com.mayabot.nlp.segment.reader

import com.google.inject.Inject
import com.google.inject.Singleton
import com.mayabot.nlp.MynlpEnv

const val StopWordDictPath = "stopword-dict/stopwords.txt"

/**
 * 停用词词典
 *
 * @author jimichan
 */
@Singleton
class StopWordDict
@Inject constructor(env: MynlpEnv) {

    private val stopWords: Set<String>

    init {

        val resource = env.loadResource(StopWordDictPath)
                ?: throw RuntimeException("Not found $StopWordDictPath Resource")

        stopWords = resource.openInputStream().bufferedReader()
                .lineSequence()
                .map {
                    it.trim()
                }.filter { it.isNotBlank() }
                .toSet()
    }

    fun getStopWords(): Set<String> {
        return stopWords
    }
}
